presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -81,7 +84,18 @@ private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics, Type
                 averageRowSize = DEFAULT_DATA_SIZE_PER_COLUMN;
             }
         }
-        return numberOfNonNullRows * averageRowSize;
+
+        double outputSize = numberOfNonNullRows * averageRowSize;
+
+        // account for "is null" boolean array
+        outputSize += outputRowCount * Byte.BYTES;
+
+        // account for offsets array for variable width types
+        if (type instanceof VariableWidthType) {
```

Comment:
Do you think we should have a method in the `Type` interface that returns an anticipated storage overhead for a single value?

Reply:
This estimate here is pessimistic one (but probably best we have for now). One cannot assume that data will be stored on a specific block type (and this could differ at runtime). Data could be stored in dictionary blocks or be RLE encoded. Therefore I think type is not the best place to provide anticipated overhead.

I think in the future we could extend the code here to guess what type of block would be used based on plan structure.

Reply:
We could also query type for block builder to get overhead from it